### PR TITLE
feat(tracing): Add config options

### PR DESCRIPTION
**Pull request by relaxolotl**
This adds basic config values to sentry options that will be used by performance monitoring. The traces sampler is deliberately omitted from this due to its complexity.

### CODE REVIEW

**Comment by loewenheim** [comment]
Anecdotally, tests work on my Macbook.

**Comment by relaxolotl** [comment]
> Anecdotally, tests work on my Macbook.

It turned out that crashpad was on some other commit on my local machine, likely for some other testing that I did in the past 🙃 I reverted that and the unit tests looked good, thank you greatly for the sanity check on this.

This PR is soft-blocked by https://github.com/getsentry/sentry-native/pull/615 because of an issue with CI; the linux codechecker step will consistently fail until that PR is merged in.

cc @loewenheim

**Review by Swatinem — APPROVED** [review]

**Comment by relaxolotl** [comment]
For those that happen to stumble upon this PR, turning these options on does literally nothing right now.

I'm thinking maybe we could hide this and all of the performance monitoring code behind a compile time option in a follow-up PR, just to avoid any confusion.

**Pull request merged**
